ci/snap/reset-tools/snapcraft.yaml

```diff
@@ -0,0 +1,138 @@
+name: factory-reset-tools
+version: '1.0'
+summary: Factory Reset Tools
+description: |
+  A tool to create reset media and reboot into the factory reset menu, for Ubuntu-preinstalled systems.
+
+base: core22
+confinement: strict
+grade: stable
+
+# for now, only build for amd64, since PC OEM does not have other architectures at the moment...
+architectures:
+- build-on: [amd64]
+
+apps:
+  factory-reset-tools:
+    command: factory-reset-tools
+    extensions: [gnome]
+    plugs:
+      - shutdown
+      - removable-media
+      - block-devices
+      - reset-partition-fsuuid
+      - reset-yaml
+      - udisks2
+      - hardware-observe
+      - mount-observe
+      - dbus-client
+  cli:
+    command: factory-reset-tools-cli
+    plugs:
+      - shutdown
+      - removable-media
+      - block-devices
+      - reset-partition-fsuuid
+      - reset-yaml
+      - udisks2
+      - hardware-observe
+      - mount-observe
+      - dbus-client
+  dbus-daemon:
+    command: factory-reset-tools-cli dbus
+    daemon: simple
+    activates-on: [dbus-service]
+    plugs:
+      - shutdown
+      - boot-grub
+      - reset-yaml
+
+parts:
+  flutter-git:
+    plugin: nil
+    source: .
+    override-pull: |
+      # necessary when reruns using the same container
+      if [ -d "flutter" ]; then
+        rm -rf "flutter"
+      fi
+
+      craftctl default
+      FLUTTER_VERSION=$(sed -n "s/^flutter \([0-9.]\+\).*/\1/p" .tool-versions)
+      git clone -b $FLUTTER_VERSION --depth 1 https://github.com/flutter/flutter.git
+    override-build: |
+      mkdir -p $CRAFT_PART_INSTALL/usr/bin
+      mkdir -p $CRAFT_PART_INSTALL/usr/libexec
+      cp -r $CRAFT_PART_SRC/flutter $CRAFT_PART_INSTALL/usr/libexec/flutter
+      ln -s $CRAFT_PART_INSTALL/usr/libexec/flutter/bin/flutter $CRAFT_PART_INSTALL/usr/bin/flutter
+      ln -s $CRAFT_PART_INSTALL/usr/libexec/flutter/bin/dart $CRAFT_PART_INSTALL/usr/bin/dart
+      $CRAFT_PART_INSTALL/usr/bin/flutter doctor
+    build-packages:
+      - clang
+      - cmake
+      - curl
+      - libgtk-3-dev
+      - ninja-build
+      - unzip
+      - xz-utils
+      - zip
+    override-prime: ""
+
+  factory-reset-tools:
+    after: [flutter-git]
+    plugin: nil
+    source: .
+    source-type: git
+    build-attributes: [enable-patchelf]
+    override-build: |
+      set -eux
+      mkdir -p $CRAFT_PART_INSTALL/bin/lib
+      cp snap/local/launcher $CRAFT_PART_INSTALL/bin/
+      mkdir -p $CRAFT_PART_INSTALL/usr/share/applications
+      cp snap/local/factory-reset-tools.desktop $CRAFT_PART_INSTALL/usr/share/applications/
+      
+      dart pub global activate melos
+      dart pub global run melos bootstrap
+      cd packages/factory_reset_tools
+      flutter build linux --release -v
+      cp -r build/linux/*/release/bundle/* $CRAFT_PART_INSTALL/bin/
+
+  factory-reset-tools-cli:
+    after: [flutter-git]
+    source: .
+    plugin: nil
+    flutter-target: lib/main.dart
```

Comment:
```
Bad snapcraft.yaml content:
- extra field 'flutter-target' not permitted in 'parts.factory-reset-tools-cli' configuration 
```

Reply:
I just got it from here: https://github.com/canonical/factory-reset-tools/blob/main/snap/snapcraft.yaml
But it looks to me like that line might not be needed at all?

Reply:
ah, that was using snapd's flutter plugin (which I've never used, so I don't know how well this works)

Reply:
@d-loose Should we just merge this and iterate on it later, once the tools are ready?
So that the job doesn't fail.